### PR TITLE
Update the binding into IoC

### PR DESCRIPTION
**Pull request by nWidart**
This allows the injection of the Twitter class in contructors and such.

### CODE REVIEW

**Comment by thujohn** [comment]
Hi.

Is it compatible with PHP 5.4?

**Comment by nWidart** [comment]
Hey, 

Indeed this is not 5.4 compatible due to the ::class notation.

Strange, something must be expecting the `ttwitter` key somewhere, I'll check it out tomorrow.

**Comment by thujohn** [comment]
I want to keep it compatible with L4 ;)

**Comment by nWidart** [comment]
fyi, 5.4 is end of life since sep 14th, don't know if it's worth keeping support for that. You could always keep a legacy branch for l4, and move on on master.

**Comment by thujohn** [comment]
Is this notation really helpful ?

**Comment by nWidart** [comment]
Well we can always keep the old way `$this->app[Thujohn\Twitter\Twitter]`. Main thing is to have the class in there so that in can be injected into classes.

The reason `::class` is nice, is when you refactor (using your ide), it will also change places where you used `::class`, where if you used a string, the ide wouldn't know the class was used there.

**Comment by thujohn** [comment]
I didn't know I only use Sublime Text :)

So if you find the problem maybe we can add a conditional statement for >= 5.4

**Comment by joshbrw** [comment]
:-1: `::class` is a much tidier notation, and in my opinion there's no reason for not keeping up to date with the latest PHP version.

File: src/Thujohn/Twitter/TwitterServiceProvider.php [hunk 1]
```diff
@@ -62,4 +62,4 @@ public function provides()
 		return ['ttwitter'];
```
Comment (joshbrw) [inline src/Thujohn/Twitter/TwitterServiceProvider.php]: Shouldn't this be `return [Twitter::class]`?

Comment (nWidart) [inline src/Thujohn/Twitter/TwitterServiceProvider.php]: Maybe, though changing that + the facade, still gives that error.

Though it works in my project, it only gives the error when running the `ide-helper:generate` command

Comment (joshbrw) [inline src/Thujohn/Twitter/TwitterServiceProvider.php]: It probably doesn't even matter actually, as this Service Provider isn't `deferred`

Comment (nWidart) [inline src/Thujohn/Twitter/TwitterServiceProvider.php]: Yup, but changing the facade to:

``` php
protected static function getFacadeAccessor() { return '\Thujohn\Twitter\Twitter'; }
``` 
Produced a different error:

```
Exception: Undefined index: ttwitter::config
```

I'll keep it how it's setup in the PR as it's working fine in my project.

**Comment by jfadich** [comment]
When you use dependency injection, Laravel doesn't use Facade at all. That's why changing the facade access key will has not fixed the issue.

The issue here is that when the IoC goes to instantiate the Twitter class it can't because the $config variable is not injectable. I've submitted another pull request that will fix it.

**Comment by thujohn** [comment]
Added in 2.1.4

File: src/Thujohn/Twitter/TwitterServiceProvider.php (hunk 1 above)
Comment (osugregor) [inline src/Thujohn/Twitter/TwitterServiceProvider.php]: Does the PHP version need to be updated now?
I opened #146 

Would a PHP version update justify a minor version change as opposed to a patch?

Thanks,
Greg

**Pull request closed without merging**